WordPress/gutenberg
# Improve the placeholder instructions accessibility.

Fixes https://github.com/WordPress/gutenberg/issues/42738

## What?

https://github.com/WordPress/gutenberg/pull/38366 made sure the placeholder instructions are announced by screen readers. It did so by using a `<fieldset>` element with a `<legend>` element that holds the instructions text.
However, that appears to be not ideal for the reasons explained in #42738. I'd like to propose a cleaner approach.

## Why?

- A fieldset element establishes a group that groups logically related form controls. A legend element is meant to give the group a meaningful name. It's not the right place for instructions.
- Some screen readers (e.g. VoiceOver) read out the legend multiple times, each time a form control within the fieldset receives focus. That's very noisy for screen reader users.
- Some blocks (e.g. the 'Table of contents' block) do have instructions text but don't contain form controls and there's no action to take. The fieldset and legend are rendered regardless. That's a named group that doesn't contain anything.
- Other blocks don't contain form controls depending on their state, see https://github.com/WordPress/gutenberg/issues/42738#issuecomment-1315428420. Also in this case a fieldset and legend aren't ideal.
- Lastly, third-party blocks may provide a placeholder with any kind of content, even with forms and fieldsets, which would result in nested fieldsets. While that's allowed in HTML, it's problematic for assistive technologies.

## How?

The only goal of the fieldset and legend added in 38366 was to make the instructions announced by screen readers. It's safe to remove them and provide another way to make screen readers announce the instructions. What this PR does:
- Removes the fieldset and legend.
- Adds an ARIA `role=status` to the instructions so that they're announced just once, when inserting the block.
- Adjusts the CSS.
- Adjusts the tests accordingly.

Note: a `role=status` establishes a live region, that is a region whose content is supposed to be updated. Strictly speaking the usage ot `role=status` in this case is slightly unsemantic because the instructions text doesn't update.

## Testing Instructions




- Test with a screen reader e.g. Safari + VoiceOver (on macOS) or Firefox + NVDA (on Windows).
- Edit a post or create a new one.
- Insert a block that has a Placeholder with instructions, e.g. an Image block.
- Check that the screen reader announces the instructions.
- Add a block that doesn't have form controls within the Placeholder, e.g. a Table of Contents block.
- Check that the screen reader announces the instructions.

Without a screen reader:
- Check the instructions text container has a `role="status"` attribute.

## Screenshots or screencast
